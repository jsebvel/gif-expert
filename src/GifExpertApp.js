import { useState } from "react";
import { AddCategory } from "./Components/AddCategory";
import { GifGrid } from "./Components/GifGrid";

const GifExpertApp = ({defaultCategories = []}) => {
    // const [categories, setCategories] = useState(['One punch']);
    const [categories, setCategories] = useState(defaultCategories);
    // const categories = ['One punch', 'Samuari X', 'Dragon Ball'];
    // const handleApp = () => {
    //     setCategories([...categories, 'Yuyu Hakusho']);
    //     console.log(categories);
    // }

    return <>
        <h2>Gif ExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr />
        <ol>
            {
                categories.map(categorie => (
                   <GifGrid
                    key={categorie}
                    category= {categorie}
                    />
                ))
            }
        </ol>
    </>
}
export default GifExpertApp;