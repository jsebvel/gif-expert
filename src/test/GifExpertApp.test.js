import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';
import React from 'react';


describe('Testing on Gif expert app', () => {

    test('should match with snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show category list', () => {
        const categories = ['goku', 'saitama', 'the hunter'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
});