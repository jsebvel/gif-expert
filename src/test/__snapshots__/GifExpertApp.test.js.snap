// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on Gif expert app Should show category list 1`] = `
<Fragment>
  <h2>
    Gif ExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="goku"
      key="goku"
    />
    <GifGrid
      category="saitama"
      key="saitama"
    />
    <GifGrid
      category="the hunter"
      key="the hunter"
    />
  </ol>
</Fragment>
`;

exports[`Testing on Gif expert app should match with snapshot 1`] = `
<Fragment>
  <h2>
    Gif ExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
